.github/workflows/main.yml

```diff
@@ -627,26 +654,45 @@ jobs:
         fi
 
     # Build and test all features
-    - run: ./ci/run-tests.sh ${{ matrix.extra_features }} --locked
+    - run: ./ci/run-tests.sh ${{ matrix.extra_features }} ${{ matrix.bucket }} --locked
       env:
         RUST_BACKTRACE: 1
 
-    # Test debug (DWARF) related functionality.
+    # NB: the test job here is explicitly lacking in cancellation of this run if
+    # something goes wrong. These take the longest anyway and otherwise if
+    # Windows fails GitHub Actions will confusingly mark the failed Windows job
+    # as cancelled instead of failed.
+
+
+  # Test debug (DWARF) related functionality.
+  test_debug_dwarf:
```

Comment:
I think this job is worth splitting out regardless if we want the rest of this PR. This is an extra 4-5 minutes at the end of a full `ci/run-tests.sh` on `main`.